src/hotspot/share/gc/z/zStat.cpp

```diff
@@ -1544,79 +1558,90 @@ void ZStatRelocation::print_page_summary() {
   lt.print("Forwarding Usage: " SIZE_FORMAT "M", _forwarding_usage / M);
 }
 
-void ZStatRelocation::print_age_table() {
+void ZStatRelocation::print_age_table() const {
   LogTarget(Info, gc, reloc) lt;
+
   if (!_selector_stats.has_relocatable_pages() || !lt.is_enabled()) {
     // Nothing to log or logging not enabled.
     return;
   }
 
+  // Find range of pages to print
+  const uint end = [&]() {
+    for (uint i = 0; i <= ZPageAgeMax; ++i) {
+      const ZPageAge age = static_cast<ZPageAge>(i);
+      if (age == ZPageAge::old) {
+        return i;
+      }
+
+      if (_selector_stats.npages(age) == 0 && _previous_selector_stats.npages(age) == 0) {
+        return i;
+      }
+    }
+
+    return ZPageAgeMax;
+  }();
```

Comment:
There can be cases where there are empty ages in the middle, I think it pretty much only happens when `ZTenuringThreshold` is used. So maybe reverse iterate from `ZPageAgeMax-1` and find the first where some data is not 0 and return `i + 1`